src/stan/mcmc/hmc/base_hmc.hpp

```diff
@@ -132,8 +132,18 @@ class base_hmc : public base_mcmc {
     this->z_.ps_point::operator=(z_init);
   }
 
+  /**
+   * The mass matrix.
```

Comment:
Sorry.  I should've been more specific.  

This should follow our genreal standards for Doxygen doc and ideally also use full sentences in sentence contexts.

So this should be:

```
/**
 * Return the representation of a point in phase space.
 *
 * @tparam Model a Stan model
 * @tparam BaseRNG base random number generator type
 * @return point in phase space
 */
```

`z_` here is the triple of `q` (position), `p` (velocity), and `g` (I don't know what that is).   I believe `g` may be the metric we're using here, since the function `PointType` has a `write_metric` method.  We've bee moving to "metric" from "mass matrix" to keep everything consistent.

That way the template parameters and return get documented.  I

Reply:
The class template parameters (`Hamiltonian`, `Model`, `BaseRNG`) should probably be documented on the class rather than on its methods, so I'm adding those comments there.
I'm not sure `base_hmc` is the right place to explain the nature of a *point* in the context of HMC, it should go in `ps_point` and `{unit|diag|dense}_e_point`. I think we could reference the [NUTS paper](https://arxiv.org/pdf/1111.4246.pdf) or some other documentation and make sure we use the same nomenclature for symbol names -- the term *phase space* does not appear in that paper so I used *(unconstrained) parameter space*, but of course happy to change to anything else.
Would this work?

```cpp
/**
 * Base class for Hamiltonian samplers.
 *
 * @tparam Model The type of the Stan model.
 * @tparam Hamiltonian The type of Hamiltonians over the (unconstrained)
 * parameter space.
 * @tparam Integrator The type of integrator (e.g. leapfrog).
 * @tparam BaseRNG The type of random number generator.
 */
 template <class Model, template <class, class> class Hamiltonian,
          template <class> class Integrator, class BaseRNG>
class base_hmc : public base_mcmc {

 ...

  /**
   * Gets the current point in the (unconstrained) parameter space.
   *
   * @return The current point in the (unconstrained) parameter space.
   */
  typename Hamiltonian<Model, BaseRNG>::PointType& z() { return z_; }

  /**
   * Gets the current point in the (unconstrained) parameters space.
   *
   * @return The current point in the (unconstrained) parameters space.
   */
  const typename Hamiltonian<Model, BaseRNG>::PointType& z() const noexcept {
    return z_;
  }

...
```

I believe `g` is the precomputed gradient for the Hamiltonian, see the `dphi_dq` methods in the `{unit|diag|dense}_e_metric` classes, and the comment in the `base_hamiltonian` class `// phi = 0.5 * log | Lambda (q) | + V(q)`